lib/readline.js

```diff
@@ -47,9 +47,9 @@ function Interface(input, output, completer, terminal) {
   }
   historySize = historySize || kHistorySize;
 
-  completer = completer || function() { return []; };
+  completer = completer || undefined;
 
-  if (typeof completer !== 'function') {
+  if (! typeof completer in ['function', 'undefined']) {
```

Comment:
Heh. Guess I should add a test for this line because mine doesn't work at all. Should be using `indexOf()` for test against the Array rather than `in`. Does the inline/optimization advice still apply if I'm doing `['function', 'undefined'].indexOf(typeof completer) === -1`?